### ISSUE TITLE
Non-ASCII tokens are corrupted sometimes when using the streaming API

### ISSUE THREAD

**Issue by hvenev-insait** [openai/openai-node#706]
### Confirm this is a Node library issue and not an underlying OpenAI API issue

- [x] This is an issue with the Node library

### Describe the bug

When using the streaming API, sometimes tokens get corrupted. Characters get replaced by two or more `\uFFFD`. For example:

```
{
  choices: [ { text: ' из��естни' } ],
}
```

when the token received is actually supposed to be `' известни'`.

The issue occurs because `LineDecoder` does not deal with multi-byte characters on chunk boundaries. Instead of using a separate `TextDecoder` instance per buffer, perhaps it should use a single `TextDecoderStream` for the entire stream.

### To Reproduce

1. Send a streaming completion request that will get non-ASCII tokens as a response.
2. Observe the output. With some probability, some of the tokens will be corrupted.

### OS

Linux

### Node version

Node v18.19.1

### Library version

openai v4.14.2

**Comment by rattrayalex** [comment]
cc @robertcraigie

**Comment by rattrayalex** [comment]
Thank you for the report! We can reproduce this and expect to have a fix out in the coming days. Thanks for your patience.

**Comment by benchaikin** [comment]
Any update on this?

**Comment by rattrayalex** [comment]
Sorry, we've been a bit delayed here - we hope to take another crack tomorrow.

**Comment by benchaikin** [comment]
@rattrayalex  Any progress on this or any way I can possibly help?

**Comment by RobertCraigie** [comment]
Sorry I forgot to update here, this should be fixed now! Have you tried the latest version?

**Comment by benchaikin** [comment]
Amazing! I'll snag the latest version and put it through its paces. Thanks so much!

**Comment by benchaikin** [comment]
Just tested out v4.33.0. Still seeing unknown characters for:

- Completion calls
- Tools_call function arguments
- Transcription calls

Example response: "Claro, me gustar�a saber sobre una situaci�n espec�fica..."

**Comment by RobertCraigie** [comment]
Can you share an example script to reproduce that?

**Comment by benchaikin** [comment]
Here's a silly example: 
`    const openAi = new OpenAI({ apiKey });
    const response = await openAi.chat.completions.create({
      model: 'gpt-4-1106-preview',
      messages: [
        { role: 'assistant', content: 'Name a topic' },
        { role: 'user', content: 'The sun.' },
      ],
      tools: [
        {
          type: 'function',
          function: {
            name: 'limerick_func',
            description:
              'returns a limerick in Spanish when the USER asks about a topic',
            parameters: {
              type: 'object',
              properties: {
                limerick: {
                  type: 'string',
                  description: 'The contents of the Spanish limerick',
                },
              },
              required: ['limerick'],
            },
          },
        },
      ],
      stream: false,
    });

    console.log(response.choices[0].message.tool_calls[0].function.arguments);`

Output: `{"limerick":"En el cielo un astro se ve,\nque alumbra con fuerza y fe,\nel sol sin igual,\nda luz y calor vital,\ny en el d�a su poder se ve."}`

**Comment by RobertCraigie** [comment]
Thanks, ran that a couple of times and didn't see any weird characters.

That script also isn't using streaming? So I suspect something else is happening in your case

**Comment by rattrayalex** [comment]
A reliable repro would be most useful!

**Comment by benchaikin** [comment]
"Reliable" is the tricky bit here as the issue isn't consistent. However after running the previous script repeatedly I'm able to get invalid characters in the function arguments:

`{"limerick":"En el cielo brilla el sol,\ncon sus rayos da calor,\nilumina el d�a entero,\nes de fuego un gran lucero,\nsu belleza es sin igual, un esplendor."}`

as well as the occasional unicode escape characters:

`{"limerick":"En el cielo brilla el sol,\ncon su luz da inspiraci\\u00f3n,\nilumina el d\\u00eda,\ncon su energ\\u00eda,\ny nos da mucha emoci\\u00f3n."}`

**Comment by rattrayalex** [comment]
ok thanks, we'll try to take a look next week!

**Comment by RobertCraigie** [comment]
@benchaikin thanks for the repro, I did manage to reproduce the issue you're seeing but unfortunately it does not appear to be an SDK issue.

I could also reproduce this with the Python SDK and the API is responding with the binary data for the � symbol directly, it's not a decoding issue. i.e. it's responding with the binary data `\xef\xbf\xbd` which when decoded, turns into �.

I've passed this on to the OpenAI team.
